### DISCUSSION TITLE
Inconsistent naming convention NEXTCLOUD_DATADIR vs NEXTCLOUD_DATA_DIR

### DISCUSSION THREAD

**Discussion by 5hanth** [nextcloud/all-in-one#2108] [Questions] [answered]
Hi I find there are multiple versions of variable `NEXTCLOUD_DATADIR` and `NEXTCLOUD_DATA_DIR` are used in this repo. I believe this is affecting my installation. I have mounted Digital ocean spaces using s3fs to `/media/infra-space`

if I use `NEXTCLOUD_DATADIR`, there are write permission errors for www-data user as the shell scripts don't seem to read this and apply correct permissions so next cloud fails to start its container. 

if I use `NEXTCLOUD_DATA_DIR`, everything works; just that the location is not what's given but a default path `/var/lib/docker/volumes/nextcloud_aio_nextcloud_data/_data/` 

is my assumption correct here or am I missing something else to ensure files are stored in mounted s3fs path of instance?

`docker inspect nextcloud_aio_nextcloud`

```
      "HostConfig": {
            "Binds": [
                "nextcloud_aio_nextcloud:/var/www/html:rw",
                "nextcloud_aio_nextcloud_data:/mnt/ncdata:rw",
                "/media/infra-space/:/media/infra-space/:rw"
            ],
            ...
      }
      ...
      
      
          "Mounts": [
            {
                "Type": "volume",
                "Name": "nextcloud_aio_nextcloud",
                "Source": "/var/lib/docker/volumes/nextcloud_aio_nextcloud/_data",
                "Destination": "/var/www/html",
                "Driver": "local",
                "Mode": "rw",
                "RW": true,
                "Propagation": ""
            },
            {
                "Type": "volume",
                "Name": "nextcloud_aio_nextcloud_data",
                "Source": "/var/lib/docker/volumes/nextcloud_aio_nextcloud_data/_data",
                "Destination": "/mnt/ncdata",
                "Driver": "local",
                "Mode": "rw",
                "RW": true,
                "Propagation": ""
            },
            {
                "Type": "bind",
                "Source": "/media/infra-space",
                "Destination": "/media/infra-space",
                "Mode": "rw",
                "RW": true,
                "Propagation": "rprivate"
            }
        ],
      
      ...
      
      
      "Env": [
           "NEXTCLOUD_MOUNT=/media/infra-space/",
           ..
           ]
      
```

**Comment by szaimen** [comment, marked as answer]
Hi, no NEXTCLOUD_DATADIR is correct. I suppose the initial installation is failing on you because you did not apply the correct permission to the target directory. This will be improved with https://github.com/nextcloud/all-in-one/pull/2109
